src/Microsoft.Azure.ServiceBus/Primitives/SecurityToken.cs

```diff
@@ -4,79 +4,59 @@
 namespace Microsoft.Azure.ServiceBus.Primitives
 {
     using System;
-    using System.Collections.Generic;
-    using System.Globalization;
-    using System.Net;
 
     /// <summary>
     /// Provides information about a security token such as audience, expiry time, and the string token value.
     /// </summary>
-    internal class SecurityToken
+    public class SecurityToken
     {
-        // per Simple Web Token draft specification
-        private const string TokenAudience = "Audience";
-        private const string TokenExpiresOn = "ExpiresOn";
-        private const string TokenIssuer = "Issuer";
-        private const string TokenDigest256 = "HMACSHA256";
+        /// <summary>
+        /// Token literal
+        /// </summary>
+        protected string token;
 
-        const string InternalExpiresOnFieldName = "ExpiresOn";
-        const string InternalAudienceFieldName = TokenAudience;
-        const string InternalKeyValueSeparator = "=";
-        const string InternalPairSeparator = "&";
-        static readonly Func<string, string> Decoder = WebUtility.UrlDecode;
-        static readonly DateTime EpochTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
-        readonly string token;
-        readonly DateTime expiresAtUtc;
-        readonly string audience;
+        /// <summary>
+        /// Expiry date-time
+        /// </summary>
+        protected DateTime expiresAtUtc;
 
         /// <summary>
-        /// Creates a new instance of the <see cref="SecurityToken"/> class.
+        /// Token audience
         /// </summary>
-        /// <param name="expiresAtUtc">The expiration time</param>
-        public SecurityToken(string tokenString, DateTime expiresAtUtc, string audience)
-        {
-            if (string.IsNullOrWhiteSpace(tokenString))
-            {
-                throw Fx.Exception.ArgumentNull(nameof(tokenString));
-            }
-            if (string.IsNullOrWhiteSpace(audience))
-            {
-                throw Fx.Exception.ArgumentNull(nameof(audience));
-            }
+        protected string audience;
 
-            this.token = tokenString;
-            this.expiresAtUtc = expiresAtUtc;
-            this.audience = audience;
+        /// <summary>
+        /// Token type
+        /// </summary>
+        protected string tokenType;
+
+        internal SecurityToken()
```

Comment:
Is empty internal ctor needed?

Reply:
Removed this, derived classes can call the default constructor now.

---
In reply to: [156531623](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156531623) [](ancestors = 156531623)